example/llm/sagemaker_deploy.ipynb

```diff
@@ -0,0 +1,368 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use Amazon SageMaker to deploy a model from the Hugging Face Hub\n",
+    "\n",
+    "### Before running the code\n",
+    "\n",
+    "You will need a valid [AWS CLI profile](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) to run the code. You can set up the profile by running `aws configure --profile <profile_name>` in your terminal. You will need to provide your AWS Access Key ID and AWS Secret Access Key. You can find your AWS Access Key ID and AWS Secret Access Key in the [Security Credentials](https://console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials) section of the AWS console.\n",
+    "\n",
+    "```bash\n",
+    "$ aws configure --profile <profile_name>\n",
+    "$ AWS Access Key ID [None]: <your_access_key_id>\n",
+    "$ AWS Secret Access Key [None]: <your_secret_access_key>\n",
+    "$ Default region name [None]: us-west-2\n",
+    "$ Default output format [None]: .json\n",
+    "```\n",
+    "\n",
+    "We recommend using the default profile by executing the `aws configure` command. This notebook will utilize the default profile. Make sure to set `Default output format` to `.json`.\n",
+    "\n",
+    "> Note: If you don't have AWS CLI installed, you will get a `command not found: aws` error. You can follow the instructions [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For more details on how to deploy a model on Amazon SageMaker, you can refer to this document:\n",
+    "\n",
+    "https://huggingface.co/docs/sagemaker/inference#deploy-a-model-from-the--hub\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Import dependency\n",
+    "First, we import libraries and create a boto3 session. We will use the default profile here, but you can also specify a profile name."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml\n",
+      "sagemaker.config INFO - Not applying SDK defaults from location: /home/ubuntu/.config/sagemaker/config.yaml\n"
+     ]
+    }
+   ],
+   "source": [
+    "import json\n",
+    "from datetime import datetime\n",
+    "\n",
+    "import boto3\n",
+    "import sagemaker\n",
```

Comment:
qq: based on this conversation on [slack](https://cambiomlworkspace.slack.com/archives/C05HZHD6YKB/p1705015889764589). Is it possible we can use boto3 instead sagemaker SDK here due to sagemaker is using very old version pydantic.

Reply:
Regarding sagemaker_deploy.ipynb, it is unlikely that we can completely remove sagemaker SDK from the notebook since we are utilizing from sagemaker.huggingface import HuggingFaceModel, get_huggingface_llm_image_uri. However, if we wish to deploy an endpoint without using sagemaker SDK, we can make use of sagemaker_deploy_mistral.ipynb, which does not necessitate the use of sagemaker SDK.

Reply:
Thanks for the clarification. As long as there is no sagemaker package import in the .py file, I think we are fine.